cmd/registry/conformance/linter_test.go

```diff
@@ -0,0 +1,292 @@
+							},
+						},
+					},
+				},
+				"spectral": {
+					name:  "spectral",
+					rules: []string{rules[1].GetLinterRulename(), rules[2].GetLinterRulename()},
+					rulesMetadata: map[string]*ruleMetadata{
+						rules[1].GetLinterRulename(): {
+							guidelineRule: rules[1],
+							guideline: &rpc.Guideline{
+								Id:     "refproperties",
+								Rules:  []*rpc.Rule{rules[0], rules[1]},
+								Status: rpc.Guideline_ACTIVE,
+							},
+						},
+						rules[2].GetLinterRulename(): {
+							guidelineRule: rules[2],
+							guideline: &rpc.Guideline{
+								Id:     "descriptionproperties",
+								Rules:  []*rpc.Rule{rules[2]},
+								Status: rpc.Guideline_PROPOSED,
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			desc: "Empty metadata",
+			styleguide: &rpc.StyleGuide{
+				Id:        "openapitest",
+				MimeTypes: []string{"application/x.openapi+gzip;version=3.0.0"},
+				Guidelines: []*rpc.Guideline{
+					{
+						Id:     "refproperties",
+						Rules:  []*rpc.Rule{},
+						Status: rpc.Guideline_ACTIVE,
+					},
+				},
+			},
+			want:    nil,
+			wantErr: true,
+		},
+		{
+			desc: "Missing linterName",
+			styleguide: &rpc.StyleGuide{
+				Id:        "openapitest",
+				MimeTypes: []string{"application/x.openapi+gzip;version=3.0.0"},
+				Guidelines: []*rpc.Guideline{
+					{
+						Id: "refproperties",
+						Rules: []*rpc.Rule{
+							{
+								Id:             "norefsiblings",
+								LinterRulename: "no-$ref-siblings",
+								Severity:       rpc.Rule_ERROR,
+							},
+						},
+						Status: rpc.Guideline_ACTIVE,
+					},
+				},
+			},
+			want:    nil,
+			wantErr: true,
+		},
+		{
+			desc: "Missing linterRuleName",
+			styleguide: &rpc.StyleGuide{
+				Id:        "openapitest",
+				MimeTypes: []string{"application/x.openapi+gzip;version=3.0.0"},
+				Guidelines: []*rpc.Guideline{
+					{
+						Id: "refproperties",
+						Rules: []*rpc.Rule{
+							{
+								Id:       "norefsiblings",
+								Linter:   "spectral",
+								Severity: rpc.Rule_ERROR,
+							},
+							rules[1],
+						},
+						Status: rpc.Guideline_ACTIVE,
+					},
+				},
+			},
+			want: map[string]*linterMetadata{
+				"spectral": {
+					name:  "spectral",
+					rules: []string{rules[1].GetLinterRulename()},
+					rulesMetadata: map[string]*ruleMetadata{
+						rules[1].GetLinterRulename(): {
+							guidelineRule: rules[1],
+							guideline: &rpc.Guideline{
+								Id: "refproperties",
+								Rules: []*rpc.Rule{
+									{
+										Id:       "norefsiblings",
+										Linter:   "spectral",
+										Severity: rpc.Rule_ERROR,
+									},
+									rules[1],
+								},
+								Status: rpc.Guideline_ACTIVE,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+
+			got, err := GenerateLinterMetadata(test.styleguide)
+
+			if test.wantErr && err == nil {
+				t.Errorf("Expected GenerateLinterMetadata() to return an error")
```

Comment:
Should this be `Fatalf`? If we expected an error and didn't find one, the test case doesn't seem like it's set up to make a useful diff later in the test.

Reply:
Makes sense.